src/main/java/org/swisspush/redisques/util/QueueStatisticsCollector.java

```diff
@@ -495,54 +495,93 @@ Future<Void> step1(RequestCtx ctx) {
-        if (numQueues > 256) {
-            log.warn(fmt1, numQueues);
-        } else {
-            log.debug(fmt1, numQueues);
-        }
-        long begRedisRequestsEpochMs = currentTimeMillis();
-        List<Future> responses = ctx.queueNames.stream()
-                .map(queue -> ctx.conn.send(Request.cmd(Command.LLEN, queuePrefix + queue)))
-                .collect(Collectors.toList());
-        CompositeFuture.all(responses).onComplete( ev -> {
-            long durRedisRequestsMs = currentTimeMillis() - begRedisRequestsEpochMs;
-            String fmt2 = "All those {} redis requests took {}ms";
-            if (durRedisRequestsMs > 3000) {
-                log.warn(fmt2, numQueues, durRedisRequestsMs);
+
+        return vertx.executeBlocking(executeBlockingPromise -> {
+            String fmt1 = "About to perform {} requests to redis just for monitoring";
+            if (numQueues > 256) {
+                log.warn(fmt1, numQueues);
             } else {
-                log.debug(fmt2, numQueues, durRedisRequestsMs);
-            }
-            if (ev.failed()) {
-                promise.fail(new Exception("Unexpected queue length result", ev.cause()));
-                return;
-            }
-            List<NumberType> queueLengthList = ev.result().list();
-            if (queueLengthList == null) {
-                promise.fail("Unexpected queue length result: null");
-                return;
+                log.debug(fmt1, numQueues);
             }
-            if (queueLengthList.size() != ctx.queueNames.size()) {
-                String err = "Unexpected queue length result with unequal size " +
-                        ctx.queueNames.size() + " : " + queueLengthList.size();
-                promise.fail(err);
-                return;
-            }
-            ctx.queueLengthList = queueLengthList;
-            promise.complete();
+            long begRedisRequestsEpochMs = currentTimeMillis();
+
+            List<WorkerThreadTask> workerThreadTaskList = new ArrayList<>();
+
+            ctx.queueNames.forEach(queueName -> workerThreadTaskList.add(new WorkerThreadTask(ctx, queueName)));
+
+            Future<List<NumberType>> startFuture = Future.succeededFuture(new ArrayList<>());
+            // chain the futures sequentially to execute tasks
+            Future<List<NumberType>> resultFuture = workerThreadTaskList.stream()
+                    .reduce(startFuture, (future, workerThreadTask) -> future.compose(previousResults -> {
+                        // perform asynchronous task
+                        return workerThreadTask.execute().compose(taskResult -> {
+                            // append task result to previous results
+                            previousResults.add(taskResult);
+                            return Future.succeededFuture(previousResults);
+                        });
+                    }), (a, b) -> Future.succeededFuture());
```

Comment:
I have a hard time to understand what this code does due to usage of java8 stream api.
Therefore I'll assume it does what it should (*whatever that means*).

```java
Future<List<NumberType>> resultFuture = workerThreadTaskList.stream()
        .reduce(startFuture, (future, workerThreadTask) -> future.compose(previousResults -> {
            // perform asynchronous task
            return workerThreadTask.execute().compose(taskResult -> {
                // append task result to previous results
                previousResults.add(taskResult);
                return Future.succeededFuture(previousResults);
            });
        }), (a, b) -> Future.succeededFuture());
```